src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class MethodBuilderImpl : MethodBuilder
+    {
+        internal Type? _returnType;
+        internal Type[]? _parametersTypes;
+        private readonly ModuleBuilderImpl _module;
+
+        internal MethodBuilderImpl(string name, MethodAttributes attributes, CallingConventions callingConventions, Type? returnType,
+            Type[]? parameters, ModuleBuilderImpl module, TypeBuilderImpl declaringType)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(name);
+            ArgumentNullException.ThrowIfNull(module);
+
+            _module = module;
+            Name = name;
+            Attributes = attributes;
+            CallingConvention = callingConventions;
+            _returnType = returnType;
+            _parametersTypes = parameters;
```

Comment:
Do we need to make a local copy of the parameters array?

Reply:
It is used for populating method signatures https://github.com/dotnet/runtime/blob/98d07e6342f78bf443f82f8525d2e16a993a0b1f/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MetadataHelper.cs#L67

Reply:
The usual convention for public APIs that take arrays is to make a copy of the array if the API needs the data after the call returns.

The existing RuntimeMethodBuilder does that here: https://github.com/dotnet/runtime/blob/0b47a4dc519ada9c56d6f4d62865c2d96a56510c/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeMethodBuilder.cs#L98-L106

I meant that this API should follow the convention.